src/components/Recipes/recipesWithSearch/useRecipesWithSearch.js

```diff
@@ -0,0 +1,42 @@
+import { useEffect, useState } from "react";
+import { useGenericQuery } from "../../../Hooks";
+import { PAGINATION, RECIPES_API } from "../../../constants";
+
+export const useRecipesWithSearch = (queryKey, userId, service, urlParams) => {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [search, setSearch] = useState([]);
+  const [category, setCategory] = useState("all-recipes");
+  const { data, isLoading } = useGenericQuery(
+    queryKey,
+    { category, currentPage, userId },
+    service
+  );
+  useEffect(() => {
+    setCurrentPage(1);
+  }, [category]);
```

Comment:
I would at least extract this into a variable, and try to avoid using magic numbers - numbers that do not tell you anything about them.
This example is simple and understandable but try to take this as a rule